Modules/pyexpat.c

```diff
@@ -1631,101 +1626,140 @@ static int init_handler_descrs(void)
     return 0;
 }
 
-PyMODINIT_FUNC
-MODULE_INITFUNC(void)
+static int
+pyexpat_traverse(PyObject *module, visitproc visit, void *arg)
+{
+    pyexpat_state *state = pyexpat_get_state(module);
+    Py_VISIT(state->xml_parse_type);
```

Comment:
why not visit state->error in here?

Reply:
@shihai1991 didn't realize it is needed for exceptions.  I'll add it.